Framework/DataObjects/inc/MantidDataObjects/MDEventWorkspace.tcc

```diff
@@ -719,77 +738,180 @@ TMDE(void MDEventWorkspace)::refreshCache() {
+ *
+ * @param start :: coords of the start of the line
+ * @param end :: coords of the end of the line
+ * @param num_d :: number of dimensions
+ * @param dir :: vector of the direction of the line
+ * @param length :: the length of the line
+ * @returns :: ordered set of halfway points between box crossings
+ */
+TMDE(std::set<coord_t> MDEventWorkspace)::getBoxBoundaryBisectsOnLine(
+    const Mantid::Kernel::VMD &start, const Mantid::Kernel::VMD &end,
+    const size_t num_d, const Mantid::Kernel::VMD &dir,
+    const coord_t length) const {
+  std::set<coord_t> mid_points;
+
+  // Get the smallest box size along each dimension
+  // We'll assume all boxes are this size and filter out any boundaries
+  // which are not real later (by checking for unique box IDs)
+  std::vector<coord_t> smallest_box_sizes = this->estimateResolution();
+
+  // Next, we go through each dimension and see where the box boundaries
+  // intersect the line.
+  for (size_t d = 0; d < num_d; d++) {
+    coord_t line_start = start[d];
+    coord_t line_end = end[d];
+    coord_t box_size = smallest_box_sizes[d];
+    coord_t dir_current_dim = dir[d];
+
+    // +1 to get the last box
+    size_t num_boundaries =
+        static_cast<size_t>(ceil(std::abs(line_end - line_start) / box_size)) +
+        1;
+
+    // If the line has some component in this dimension then look for boundaries
+    // it crosses
+    if (dir[d] != 0.0) {
+      getBoundariesInDimension(start, dir, num_boundaries, length,
+                               dir_current_dim, box_size, mid_points);
+    }
+  }
+  return mid_points;
+}
+
+//----------------------------------------------------------------------------------------------
+/** Insert positions-along-the-line that lie halfway between points
+ * where the line crosses box boundaries in a single dimension of the workspace
+ * into ordered set
+ *
+ * @param start :: coords of the start of the line
+ * @param dir :: vector of the direction of the line
+ * @param num_boundaries :: maximum number of boundary crossings possible if all
+ *boxes are the size of the smallest box
+ * @param length :: the length of the line
+ * @param dir_current_dim :: component of the line direction in this dimension
+ * @param box_size :: the minimum box size in this dimension
+ * @param mid_points :: ordered set of halfway points between box crossings
+ */
+TMDE(void MDEventWorkspace)::getBoundariesInDimension(
+    const Mantid::Kernel::VMD &start, const Mantid::Kernel::VMD &dir,
+    const size_t num_boundaries, const coord_t length,
+    const coord_t dir_current_dim, const coord_t box_size,
+    std::set<coord_t> &mid_points) const {
+  Mantid::Kernel::VMD lastPos = start;
+  coord_t lastLinePos = 0;
+  coord_t previousLinePos = 0;
+  coord_t line_pos_of_box_centre = 0;
+  const API::IMDNode *box = nullptr;
+  size_t last_id = static_cast<size_t>(-1);
+
+  for (size_t i = 1; i <= num_boundaries; i++) {
+    size_t current_id = static_cast<size_t>(-1);
```

Comment:
Yes it wraps to the maximum value of `size_t`. It ensures that the first box id that is found will be different to `current_id`.